js/client/modules/@arangodb/testsuites/upgrade-data.js

```diff
@@ -0,0 +1,253 @@
+/* jshint strict: false, sub: true */
+/* global print */
+'use strict';
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2017 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2014 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License")
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Dan Larkin-York
+////////////////////////////////////////////////////////////////////////////////
+
+const _ = require('lodash');
+const fs = require('fs');
+const pu = require('@arangodb/process-utils');
+const tu = require('@arangodb/test-utils');
+const toArgv = require('internal').toArgv;
+
+const optionsDocumentation = [
+  '   - `upgradeDataPath`: path to directory containing upgrade data archives'
+];
+
+const compareSemVer = (a, b) => {
+  if (a === b) {
+    return 0;
+  }
+  let lex = false;
+  const partsA = a.split('-')[0].split('.');
+  const partsB = b.split('-')[0].split('.');
+  if (partsA.length < 2 ||
+    partsA.length > 4 ||
+    partsB.length < 2 ||
+    partsB.length > 4 ||
+    !partsA.every(p => isNaN(Number(p))) ||
+    !partsB.every(p => isNaN(Number(b)))) {
+    return (a < b) ? -1 : 1;
+  }
+
+  for (let i = partsA.length; i < 4; i++) {
+    partsA.push_back("0");
+  }
+  for (let i = partsB.length; i < 4; i++) {
+    partsB.push_back("0");
+  }
+
+  for (let i = 0; i < 4; i++) {
+    const numA = Number(partsA[i]);
+    const numB = Number(partsB[i]);
+    if (numA < numB) {
+      return -1;
+    }
+    if (numB < numA) {
+      return 1;
+    }
+  }
+};
+
+const byMinimumSuportedVersion = (version) => {
+  return (testCase) => {
+    let supported = true;
+    testCase.substring(0, testCase.length - 3).split('-').map((s) => {
+      if (s.startsWith("msv")) {
+        const msv = s.substring(3);
+        if (compareSemVer(msv, version) > 0) {
+          supported = false;
+        }
+      }
+    });
+    return supported;
+  };
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// set up the test according to the testcase.
+////////////////////////////////////////////////////////////////////////////////
+const unpackOldData = (engine, version, options, serverOptions) => {
+  const archiveName = `upgrade-data-${version}-${engine}`;
+  const dataFile = fs.join(options.upgradeDataPath,
+    'data',
+    `${archiveName}.tar.gz`);
```

Comment:
@Luschgy does this work on the windows nodes?

Reply:
AFAIK, `tar.exe` is preinstalled on Windows nodes.